e2e-tests/src/main.rs

```diff
@@ -33,6 +34,53 @@ fn init_env() {
     env_logger::init();
 }
 
+/// Runs default test cases in sequence.
+fn run_default_test_cases(possible_test_cases: HashMap<&str, TestCase>, config: &Config) -> anyhow::Result<()> {
+    let _ = possible_test_cases
+        .iter()
+        .map::<anyhow::Result<()>, _> (
+            |(test_name, &test_case)| {
+                run_test_case(test_case, test_name, config)?;
+                Ok(())
+            }
+        ).collect::<Vec<_>>();
+    Ok(())
+}
+
+/// Runs specified test cases in sequence.
+/// Checks whether each provided test case is valid.
+fn run_specified_test_cases(test_cases: Vec<String>, possible_test_cases: HashMap<&str, TestCase>, config: &Config) -> anyhow::Result<()> {
+    let _ = test_cases
+        .iter()
```

Comment:
instead of `iter -> map -> collect` use for_each or simple for loop.

Reply:
OK, reworked. I had some trouble here with _Result_ in a closure, hence the _map_, but now _for_each_ seems to work fine.

Reply:
Sorry for the confusion, actually I would go even further and just use simple for loop syntax

Reply:
No problem, in the end I was wrong and _for_each_ didn't do the trick, so I went with a simple _for_ loop. Will ping you once all the pipelines pass.

Reply:
@kostekIV The e2e pipeline went green, you can re-review at your convenience.